File: LeetCode/1624. Largest Substring Between Two Equal Characters/solution.cpp
/**
 * 54 / 54 test cases passed.
 * Runtime: 0 ms
 * Memory Usage: 6.3 MB 
 */
class Solution {
public:
    int maxLengthBetweenEqualCharacters(string s) {
        unordered_map<char, int> rec;
        int ans = -1;
        for (int i = 0; i < s.size(); ++ i) {
            if (rec.count(s[i])) {
                ans = max(ans, i - rec[s[i]] - 1);
            } else {
                rec[s[i]] = i;
            }
        }
        return ans;
    }
};

File: LeetCode/1624. Largest Substring Between Two Equal Characters/solution.py
"""
54 / 54 test cases passed.
Runtime: 36 ms
Memory Usage: 15 MB
"""
class Solution:
    def maxLengthBetweenEqualCharacters(self, s: str) -> int:
        rec = {}
        ans = -1
        for i, c in enumerate(s):
            if c in rec:
                ans = max(ans, i - rec[c] - 1)
            else:
                rec[c] = i    
        return ans 
        